src/main/java/core/basesyntax/SalaryInfo.java

```diff
@@ -1,7 +1,31 @@
 package core.basesyntax;
 
+import java.time.LocalDate;
+import java.time.format.DateTimeFormatter;
+
 public class SalaryInfo {
+    private final DateTimeFormatter formatter = DateTimeFormatter.ofPattern("dd.MM.yyyy");
+
     public String getSalaryInfo(String[] names, String[] data, String dateFrom, String dateTo) {
-        return null;
+        LocalDate to = LocalDate.parse(dateTo, formatter);
+        LocalDate from = LocalDate.parse(dateFrom, formatter);
+        StringBuilder stringBuilder = new StringBuilder();
+        stringBuilder.append("Report for period ").append(dateFrom).append(" - ").append(dateTo);
+
+        for (String name : names) {
+            int salary = 0;
+            for (String day : data) {
+                String[] info = day.split(" ");
+                String[] date = info[0].split("\\.");
+                LocalDate workDay = LocalDate.of(Integer.parseInt(date[2]),
+                        Integer.parseInt(date[1]), Integer.parseInt(date[0]));
+                if (info[1].equals(name) & (workDay.isEqual(from) | workDay.isEqual(to)
```

Comment:
why do we use & and not &&. What is the difference between them ? 
same with | vs ||

Reply:
I already figured it out